### DISCUSSION TITLE
[2.3] osquery adding windows 10 machine

### DISCUSSION THREAD

**Discussion by tommorgan365** [Security-Onion-Solutions/securityonion#1735] [Unsupported Versions] [answered]
Distributed environment manager, 2 query, one fleet.  Downloaded the MSI to a windows client and ran the install.  I do not see it register in fleet.  event log orginally showed it could not find the hostname -- added it to hosts file, removed msi, deleted file structure and re-installed.  

Working on trouble shooting - from the windows client should I be able to telnet to the fleet server using port 8090?  Event viewer shows x,x,x,x:8090: i/o timeout\""  where x.x.x.x is the ip address of my fleet server.  I did run so-allow on manager to add the windows client as osquery.  Did I miss a step?

**Comment by tommorgan365** [comment]
sudo salt-call pillar.get global:url_base show the mangers IP and the MSI is trying to go the the fleet's ip.

**Reply by tommorgan365** [reply]
TransientFailure, latest connection error: connection error: desc = \"transport: authentication handshake failed: x509: certificate is valid for somanager01, not sofleet01\""  -- this looks like an issue with distributed environment?

**Comment by tommorgan365** [comment, marked as answer]
checking the firewall on the sofleet machine -

I do not see the address range I added with so-allow on the somanager for osquery.

Chain DOCKER-USER (1 references)
target     prot opt source               destination
ACCEPT     tcp  --  sofleet        anywhere             tcp dpt:8090
ACCEPT     tcp  --  x.x.x.x          anywhere             tcp dpt:8090
ACCEPT     tcp  --  x.x.x.x          anywhere             tcp dpt:8090
ACCEPT     tcp  --  somanager      anywhere             tcp dpt:8090
ACCEPT     tcp  --  sofleet        anywhere             tcp dpt:8090

**Reply by tommorgan365** [reply]
I ran so-allow on the fleet manager and added my windows 10 machine - everything seems to be working now.  So my guess is you still need to run so-allow on the fleet machine separately from the manager.

**Comment by defensivedepth** [comment]
@tommorgan365 I know that you already got it working, but would you mind describing your setup a little more? Did you initially have Fleet on the Manager, then, you stood up a Standalone Fleet Node? Not sure what you meant when you say you have `2 query, one fleet` ?

**Comment by tommorgan365** [comment]
Hello
Four severs - one manager ,two search nodes and a separate fleet node.  Started out with fleet node.  My confusion is where you run the so-allow when you have a separate fleet node.  Started running it on the manager - then realized the firewall rules were not being pushed to the fleet server.  I had to add os query to the fleet server using so- allow.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by defensivedepth** [reply]
Which server did you install first? The Manager or the Fleet node?

**Comment by tommorgan365** [comment]
The manager was first.


Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by tommorgan365** [reply]
Is that the correct order with a separate fleet server ?

**Reply by defensivedepth** [reply]
Yep. Will get this further documented this week.
